### DISCUSSION TITLE
Any advice for multiple up/downlinks?

### DISCUSSION THREAD

**Discussion by JackNWhite** [nasa/fprime#1311] [Q&A]
_EDIT: This first diagram was not clear. Please skip to the second diagram below._

Good afternoon all,

The system on which I am working with F' is one of two computers which are subordinate to another, external platform. There are direct links between each of the two computers and the platform, and another link between the F' computer and its sister (let's call it !F').
![MM_Diagram](https://user-images.githubusercontent.com/10062543/157026888-958bfff9-85af-4984-9809-d1367b4bf7f0.png)

How might one approach setting this up in F'? Has anyone done it? My feeling is that you would have to modify TlmChan and CmdDispatcher to have a queue per link.

Do you have any deep thoughts on this issue?

**Comment by Emil808** [comment]
I'm trying to understand this diagram. There are three machines, two of which are connected to Platform. The same two are also running a deployment of F'. The two F' deployments, named F' and !F' are sending their telemetry/events and receiving commands from Platform. Also, F' and !F' are connected to each other. 

I'm attempting to get two F' deployments to connect with each other using the Generic Hub and TCP server/client. and am stuck in getting the IP addresses and Sockets to work together. #1305

**Reply by JackNWhite** [reply]
Sorry, I have not explained well. I try again below.

**Comment by JackNWhite** [comment]
The system on which I am working with F' is one of two computers (**Prime** and **RTOS**) which are subordinate to another, external **Platform**. There are three UARTs connecting them:
- Platform <--> Prime
- Platform <--> RTOS
- RTOS <--> Prime
![MM_Diagram2](https://user-images.githubusercontent.com/10062543/158394624-dcdc852d-b382-4148-9e79-3767f29fb981.png)

The links between the two subordinate computers and **Platform** each serve as the primary TM/TC path for each subordinate computer. However, for **Prime** there is a secondary TM/TC path directly between **Prime** and **RTOS**, so that TM/TC can pass directly between them, or between **Platform** and **Prime** via **RTOS**.

This means that **Prime** must be able to send TM packets both to **Platform** and to **RTOS** and accept TCs from both **Platform** and **RTOS**.

Has anyone done that?
